source/server/config/http/ip_tagging.cc

```diff
@@ -1,20 +1,42 @@
 #include "server/config/http/ip_tagging.h"
 
-#include <string>
-
+#include "envoy/config/filter/http/ip_tagging/v2/ip_tagging.pb.validate.h"
 #include "envoy/registry/registry.h"
 
+#include "common/config/filter_json.h"
 #include "common/http/filter/ip_tagging_filter.h"
-#include "common/json/config_schemas.h"
+#include "common/protobuf/utility.h"
 
 namespace Envoy {
 namespace Server {
 namespace Configuration {
 
 HttpFilterFactoryCb IpTaggingFilterConfig::createFilterFactory(const Json::Object& json_config,
-                                                               const std::string&,
-                                                               FactoryContext&) {
-  Http::IpTaggingFilterConfigSharedPtr config(new Http::IpTaggingFilterConfig(json_config));
+                                                               const std::string& stat_prefix,
+                                                               FactoryContext& context) {
+
+  envoy::config::filter::http::ip_tagging::v2::IPTagging proto_config;
+  Config::FilterJson::translateIpTaggingFilterConfig(json_config, proto_config);
+  return createFilter(proto_config, stat_prefix, context);
+}
+
+HttpFilterFactoryCb
+IpTaggingFilterConfig::createFilterFactoryFromProto(const Protobuf::Message& proto_config,
+                                                    const std::string& stat_prefix,
+                                                    FactoryContext& context) {
+  return createFilter(
+      MessageUtil::downcastAndValidate<
+          const envoy::config::filter::http::ip_tagging::v2::IPTagging&>(proto_config),
+      stat_prefix, context);
+}
+
+HttpFilterFactoryCb IpTaggingFilterConfig::createFilter(
+    const envoy::config::filter::http::ip_tagging::v2::IPTagging& proto_config,
+    const std::string& stat_prefix, FactoryContext& context) {
+
+  Http::IpTaggingFilterConfigSharedPtr config(new Http::IpTaggingFilterConfig(
```

Comment:
std::make_shared

Reply:
There are a few other instances of `initializer(new Type(...))` that probably also should be `std::make_shared`/`std::make_unique`.

Reply:
i'm only going to update this file. Other instances can get fixed as apart of a different pr.